### PR TITLE
Skip the webhook if not the valid data

### PR DESCRIPTION
**Pull request by ChetanGN**

### CODE REVIEW

File: includes/razorpay-webhook.php [hunk 1]
```diff
@@ -331,6 +343,19 @@ protected function getPaymentEntity($razorpayPaymentId, $data)
         return $payment;
     }
 
+    /**
+     * Returns boolean false incase not proper webhook data
+     */
+    protected function shouldConsumeWebhook($data)
+    { 
+        if (isset($data['payload']['payment']['entity']['notes']['woocommerce_order_number']) === true)
```
Comment (ramth05) [inline includes/razorpay-webhook.php:351]: @ChetanGN , let add events as well in this list with OR condition and use it once only, not with every event.

File: includes/razorpay-webhook.php [hunk 2]
```diff
@@ -331,6 +337,21 @@ protected function getPaymentEntity($razorpayPaymentId, $data)
         return $payment;
     }
 
+    /**
+     * Returns boolean false incase not proper webhook data
+     */
+    protected function shouldConsumeWebhook($data)
+    { 
+        if ((isset($data['event']) === true) and 
+            (($data['event'] === PAYMENT_AUTHORIZED) or ($data['event'] === REFUNDED_CREATED) or ($data['event'] === VIRTUAL_ACCOUNT_CREDITED) or ($data['event'] === PAYMENT_FAILED) or ($data['event'] === SUBSCRIPTION_CANCELLED)) and
```
Comment (ramth05) [inline includes/razorpay-webhook.php:346]: @ChetanGN , please consider inArray check here.

**Review by ramth05 — APPROVED** [review]
LGTM, @jampulanaveen can you please test it. @ChetanGN, Please test it properly before merge.

**Review by jampulanaveen — APPROVED** [review]
LGTM

**Pull request merged**